src/pages/resultsView/oncoprint/AddClinicalTracks.tsx

```diff
@@ -0,0 +1,228 @@
+                        return 1;
+                    } else if (clinicalAttributeIsPROFILEDIN(x)) {
+                        return 2;
+                    } else {
+                        return 3;
+                    }
+                },
+                    (x:ClinicalAttribute)=>{
+                        let freq = availableFrequency[x.clinicalAttributeId];
+                        if (freq === undefined) {
+                            freq = 0;
+                        }
+                        return -freq;
+                    },
+                    (x:ClinicalAttribute)=>-x.priority
+                    ,
+                    (x:ClinicalAttribute)=>x.displayName
+                ]
+            );
+
+            sortedAttributes.groups = _.sortBy<ClinicalAttribute>(sortedAttributes.groups, x=>x.displayName);
+
+            return Promise.resolve(_.mapValues(sortedAttributes, attrs=>{
+                return attrs.map(attr=>({
+                    label: attr.displayName,
+                    key: attr.clinicalAttributeId,
+                    selected: attr.clinicalAttributeId in this.selectedClinicalAttributeIds
+                }));
+            }));
+        }
+    });
+
+    readonly clinicalAttributeIdToAvailableFrequency = remoteData({
+        await:()=>[
+            this.props.store.clinicalAttributeIdToAvailableSampleCount,
+            this.props.store.samples
+        ],
+        invoke:()=>{
+            const numSamples = this.props.store.samples.result!.length;
+            return Promise.resolve(
+                _.mapValues(
+                    this.props.store.clinicalAttributeIdToAvailableSampleCount.result!,
+                    count=>100*count/numSamples
+                )
+            );
+        }
+    });
+
+    readonly addClinicalTracksMenu = MakeMobxView({
+        await:()=>[this.options],
+        render:()=>(
+            <AddChartByType
+                options={this.options.result!.clinical}
+                freqPromise={this.clinicalAttributeIdToAvailableFrequency}
+                onAddAll={this.addAll}
+                onClearAll={this.clear}
+                onToggleOption={this.toggleClinicalTrack}
+                optionsGivenInSortedOrder={true}
+            />
+        ),
+        renderPending:()=><LoadingIndicator isLoading={true} small={true}/>,
+        showLastRenderWhenPending:true
+    });
+
+    readonly addGroupTracksMenu = MakeMobxView({
+        await:()=>[this.options],
+        render:()=>(
+            <AddChartByType
+                options={this.options.result!.groups}
+                freqPromise={this.clinicalAttributeIdToAvailableFrequency}
+                onAddAll={this.addAll}
+                onClearAll={this.clear}
+                onToggleOption={this.toggleClinicalTrack}
+                optionsGivenInSortedOrder={true}
+            />
+        ),
+        renderPending:()=><LoadingIndicator isLoading={true} small={true}/>,
```

Comment:
i think most everywhere else we're using size attribute.  We should probably get rid of small attribute.  There's only <5 mentions

Reply:
could be another PR